core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java

```diff
@@ -126,6 +126,14 @@ public DataMapBuilder createBuilder(Segment segment, String shardName,
    */
   public Map<Segment, List<CoarseGrainDataMap>> getDataMaps(List<Segment> segments)
       throws IOException {
+    return getDataMaps(segments, null);
+  }
+
+  /**
+   * Get the datamap for all segments
+   */
+  public Map<Segment, List<CoarseGrainDataMap>> getDataMaps(List<Segment> segments,
```

Comment:
Extract common code from this and getDataMaps with single segment API

Reply:
done